### DISCUSSION TITLE
How to make web3modal use default RPC provider or WalletConnect Blockchain API before connecting to user's wallet

### DISCUSSION THREAD

**Discussion by dappbuildernet** [WalletConnect/walletconnect-monorepo#3735] [Developer Support] [answered]
It seems that there are such options in **web3modal** but no connection to blockchain created before the user connects wallet.

How can we use it with **defaultWagmiConfig**?

So that **blockchain (contracts) data can be loaded even when user has not connected wallet yet**...

I am using your example with defaultWagmiConfig that is described as having mentioned connecting options built-in.

Thank you.

**Comment by glitch-txs** [comment, marked as answer]
Hi you can use this hook
https://wagmi.sh/react/hooks/useContractRead

**Reply by dappbuildernet** [reply]
Hi, 

Thanks it is working now. 
Not sure why it did not work with viem's getContract...
